app/services/EventReportService.scala

```diff
@@ -58,15 +58,20 @@ class EventReportService @Inject()(eventReportConnector: EventReportConnector,
                                        connectToAPI: (String, String, JsValue, String) => Future[HttpResponse]
                                       )
 
-  private def apiProcessingInfo(eventType: EventType, pstr: String)
+  private def apiProcessingInfo(eventType: EventType, pstr: String, delete: Boolean)
                                (implicit headerCarrier: HeaderCarrier, ec: ExecutionContext, request: RequestHeader): Option[APIProcessingInfo] = {
     EventType.postApiTypeByEventType(eventType) flatMap {
       case Api1826 =>
-        Some(APIProcessingInfo(Api1826, API1826.transformToETMPData, SchemaPath1826, eventReportConnector.compileEventReportSummary))
+        Some(APIProcessingInfo(
+          Api1826,
+          API1826.transformToETMPData(if(delete) Some(eventType) else None),
```

Comment:
I feel it would be a little more readable & consistent if we pass in EventType and a Boolean as we do for API 1830.